formats/json-tests/commonTest/src/kotlinx/serialization/features/PolymorphismWithAnyTest.kt

```diff
@@ -117,17 +120,17 @@ class PolymorphismWithAnyTest {
     @Test
     fun testBindModules() {
         val json = Json { serializersModule = (baseAndDerivedModuleAtAny + BaseAndDerivedModule) }
-        assertStringFormAndRestored(
+        assertJsonFormAndRestored(
             expected = """{"data":{"a":{"type":"kotlinx.serialization.PolyDerived","id":1,"s":"foo"}},
```

Comment:
Perhaps we should also add a test where a slow path will be implicitly used: for example, `{"id":1,"s":"foo", "type":"kotlinx.serialization.PolyDerived"}}`

Reply:
added